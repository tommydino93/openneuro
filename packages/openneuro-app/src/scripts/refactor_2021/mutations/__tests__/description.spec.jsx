import React from 'react'
import { shallow } from 'enzyme'
import UpdateDescription, {
  mergeFieldValue,
  UPDATE_DESCRIPTION,
  UPDATE_DESCRIPTION_LIST,
} from '../description.jsx'

describe('UpdateDescription mutation', () => {
  it('renders with common props', () => {
    const wrapper = shallow(
      <UpdateDescription
        datasetId="ds001"
        field="Name"
        value="New Name"
        done={jest.fn()}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('uses the scalar mutation for scalar values', () => {
    const wrapper = shallow(
      <UpdateDescription
        datasetId="ds001"
        field="Name"
        value="New Name"
        done={jest.fn()}
      />,
    )
    expect(wrapper.find('Mutation').props().mutation).toEqual(
      UPDATE_DESCRIPTION,
    )
  })
  it('uses the list mutation for array values', () => {
    const wrapper = shallow(
      <UpdateDescription
        datasetId="ds001"
        field="Authors"
        value={['John Doe', 'Jane Doe']}
        done={jest.fn()}
      />,
    )
    expect(wrapper.find('Mutation').props().mutation).toEqual(
      UPDATE_DESCRIPTION_LIST,
    )
  })
  describe('mergeFieldValue()', () => {
    it('merges in scalar fields', () => {
      expect(
        mergeFieldValue(
          'ds001',
          {
            __typename: 'Draft',
            id: '1234',
            created: '1999',
            description: {
              Name: 'Old Name',
              BIDSVersion: '1.2',
              License: 'AGPL3',
            },
          },
          { Name: 'New Name', BIDSVersion: '1.2', License: 'AGPL3' },
          null,
        ),
      ).toEqual({
        __typename: 'Dataset',
        draft: {
          __typename: 'Draft',
          created: '1999',
          description: {
            BIDSVersion: '1.2',
            License: 'AGPL3',
            Name: 'New Name',
          },
          id: '1234',
        },
        id: 'ds001',
      })
    })
    it('merges in array fields', () => {
      expect(
        mergeFieldValue(
          'ds001',
          {
            __typename: 'Draft',
            id: '1234',
            created: '1999',
            description: {
              Name: 'Old Name',
              BIDSVersion: '1.2',
              License: 'AGPL3',
              Authors: [],
            },
          },
          null,
          {
            Name: 'New Name',
            BIDSVersion: '1.2',
            License: 'AGPL3',
            Authors: ['One', 'Two'],
          },
        ),
      ).toEqual({
        __typename: 'Dataset',
        draft: {
          __typename: 'Draft',
          created: '1999',
          description: {
            Authors: ['One', 'Two'],
            BIDSVersion: '1.2',
            License: 'AGPL3',
            Name: 'New Name',
          },
          id: '1234',
        },
        id: 'ds001',
      })
    })
  })
})
