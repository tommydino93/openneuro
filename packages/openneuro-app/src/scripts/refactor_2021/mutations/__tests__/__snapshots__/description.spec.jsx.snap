// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateDescription mutation renders with common props 1`] = `
<Mutation
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "updateDescription",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "datasetId",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "datasetId",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "field",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "field",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "value",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "value",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "updateDescription",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "Name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "BIDSVersion",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "License",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "Authors",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "Acknowledgements",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "HowToAcknowledge",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "Funding",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "ReferencesAndLinks",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "DatasetDOI",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "EthicsApprovals",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "datasetId",
                },
              },
            },
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "field",
                },
              },
            },
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "value",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 374,
        "start": 0,
      },
    }
  }
  update={[Function]}
>
  <Component />
</Mutation>
`;
